src/lib.rs

```diff
@@ -435,89 +474,81 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::remove_members(members_to_remove.len() as u32))]
 		pub fn remove_members(
 			origin: OriginFor<T>,
-			supersig_id: T::AccountId,
-			members_to_remove: Vec<T::AccountId>,
+			members_to_remove: BoundedVec<T::AccountId, T::MaxMembersPerTransaction>,
 		) -> DispatchResult {
-			let who = ensure_signed(origin)?;
-			if who != supersig_id {
-				return Err(Error::<T>::NotAllowed.into())
-			}
-			let supersig_index = Self::get_supersig_index_from_id(&supersig_id)?;
-			let mut nb_removed: usize = 0;
-
-			Supersigs::<T>::try_mutate(supersig_index, |wrapped_supersig| {
-				if let Some(supersig) = wrapped_supersig {
-					let old_len = supersig.members.len();
-					supersig.members.retain(|memb| !members_to_remove.contains(memb));
-					nb_removed = old_len - supersig.members.len();
-
-					if supersig.members.is_empty() {
-						return Err(())
-					}
-				}
-				Ok(())
-			})
-			.map_err(|_| Error::<T>::CannotRemoveUsers)?;
-			let reserve = <BalanceOf<T>>::from(size_of::<T::AccountId>() as u32)
-				.saturating_mul((nb_removed as u32).into())
-				.saturating_mul(T::PricePerBytes::get());
-			T::Currency::unreserve(&who, reserve);
-
-			Self::deposit_event(Event::<T>::UsersRemoved(supersig_id, members_to_remove));
+			let supersig_account = ensure_signed(origin)?;
+			let supersig_id = Self::get_supersig_id_from_account(&supersig_account)?;
+
+			// Remeber the storage state before we remove the members from it
+			let total_deposit = Self::total_deposit(supersig_id);
+			let initial_total_members = Self::total_members(supersig_id);
+
+			let removed_members = Self::internal_remove_members(supersig_id, members_to_remove)?;
+
+			// amount = total_deposit / initial_n_members * n_removed_members
+			let amount_to_unreserve = total_deposit
```

Comment:
This should be moved to an internal function, so we can also use it in `leave_supersig`

Reply:
Sure !